Modules/_uuidmodule.c

```diff
@@ -91,45 +111,1649 @@ py_windows_has_stable_node(void)
+    bytes_le: 'y*' = None
+    fields: object = NULL
+    int: object = NULL
+    version: object = NULL
+    *
+    is_safe: object = NULL
+
+UUID is a fast base implementation type for uuid.UUID.
+[clinic start generated code]*/
+
+static int
+_uuid_UUID___init___impl(uuidobject *self, PyObject *hex, Py_buffer *bytes,
+                         Py_buffer *bytes_le, PyObject *fields,
+                         PyObject *int_value, PyObject *version,
+                         PyObject *is_safe)
+/*[clinic end generated code: output=93a6881c8f79bf9b input=b9c79672fbd76a99]*/
+
+{
+    uuid_state *state = get_uuid_state_by_cls(Py_TYPE(self));
+
+    int passed = 0;
+    if (hex != NULL) passed++;
+    if (bytes->obj != NULL) passed++;
+    if (bytes_le->obj != NULL) passed++;
+    if (fields != NULL) passed++;
+    if (int_value != NULL) passed++;
+    if (passed != 1) {
+        PyErr_SetString(
+            PyExc_TypeError,
+            "one of the hex, bytes, bytes_le, fields, or int arguments must be given"
+        );
+        return -1;
+    }
+
+    if (hex != NULL) {
+        if (from_hex(self, hex) < 0) {
+            return -1;
+        }
+    }
+    else if (bytes->obj != NULL) {
+        if (bytes->len != 16) {
+            PyErr_SetString(
+                PyExc_ValueError,
+                "bytes is not a 16-char string"
+            );
+            return -1;
+        }
+        memcpy(self->bytes, bytes->buf, 16);
+    }
+    else if (bytes_le->obj != NULL) {
+        if (from_bytes_le(self, bytes_le) < 0) {
+            return -1;
+        }
+    }
+    else if (fields != NULL) {
+        if (from_fields(self, fields) < 0) {
+            return -1;
+        }
+    }
+    else if (int_value != NULL) {
+        if (from_int(self, int_value, 1) < 0) {
+            return -1;
+        }
+    }
+    else {
+        Py_UNREACHABLE();
+    }
+
+    if (version != NULL && version != Py_None) {
+        long version_num = PyLong_AsLong(version);
+        if (version_num == -1 && PyErr_Occurred()) {
+            return -1;
+        }
+        if (version_num < 1 || version_num > 8) {
+            PyErr_SetString(PyExc_ValueError, "illegal version number");
+            return -1;
+        }
+
+        // Clear variant bits (keep only lower 6 bits of byte 8)
+        self->bytes[8] &= 0x3f;  // 0011 1111
+        // Clear version bits (keep only lower 4 bits of byte 6)
+        self->bytes[6] &= 0x0f;  // 0000 1111
+        // Set the variant to RFC 4122/9562 (binary 10xx xxxx)
+        self->bytes[8] |= 0x80;  // 1000 0000
+        // Set the version number (upper 4 bits of byte 6)
+        self->bytes[6] |= (version_num << 4);
+    }
+
+    if (is_safe != NULL) {
+        // Validate by calling SafeUUID(is_safe) to ensure it's a valid enum member
+        PyObject *validated = PyObject_CallOneArg(state->safe_uuid, is_safe);
+        if (validated == NULL) {
+            return -1;
+        }
+        Py_CLEAR(self->is_safe);
```

Comment:
In this casey use Py_XSETREF(self->is_safe, validated) but be sure that is_safe is set. Note that with PyObject_New, the structure is not zero-initialized so you might want to be sure that you're not working on an uninitialised pointer.

Reply:
Pointers are initialized in `Uuid_alloc`. I'll update to use SETREF

Reply:
done